src/script_functions.c

```diff
@@ -116,20 +116,77 @@ static guint32 current_time(void)
 }
 
 
-static gboolean adjusting_for_decoration = FALSE;
+/**
+ * Check for the correct parameter count.
+ * Failure will log a lua error and return False.
+ * Success returns True.
+ */
+static Bool check_param_count(lua_State *lua, char *funcname, int expected)
+{
+	int top = lua_gettop(lua);
+	gchar* error_message;
 
-int c_set_adjust_for_decoration(lua_State *lua)
+	if (top != expected) {
+		error_message = num_indata_expected_errors[expected];
+		if (error_message == NULL) {
+			error_message = failed_string;
+		}
```

Comment:
If only GCC's `?:` were standard.

Reply:
This could be the full ternary instead; "error_message" isn't unwieldily long.